src/Datadog.Trace/HttpOverStreams/HttpContent/StreamContent.cs

```diff
@@ -15,14 +16,38 @@ public StreamContent(Stream stream, long? length)
 
         public long? Length { get; }
 
-        public Task CopyToAsync(Stream destination, int? bufferSize)
+        public Task CopyToAsync(Stream destination)
         {
-            if (bufferSize == null)
+            return Stream.CopyToAsync(destination);
+        }
+
+        public async Task CopyToAsync(byte[] buffer)
+        {
+            if (!Length.HasValue)
+            {
+                throw new InvalidOperationException("Unable to CopyToAsync with buffer when content Length is unknown");
+            }
+
+            if (Length > buffer.Length)
             {
-                return Stream.CopyToAsync(destination);
+                throw new ArgumentException($"Provided buffer was smaller {buffer.Length} than the content length {Length}");
             }
 
-            return Stream.CopyToAsync(destination, bufferSize.Value);
+            var length = 0;
+            var remaining = Length.Value;
```

Comment:
Given that part of the logic comes from `Length.Value` being a long, I think you should use explicit typing for this line

```suggestion
            long remaining = Length.Value;
```